c7n/resources/route53.py

```diff
@@ -208,6 +208,67 @@ def process_resource_set(self, client, domains, keys):
                 TagsToDelete=keys)
 
 
+@ResourceRecordSet.action_registry.register('delete')
+class ResourceRecordSetRemove(BaseAction):
+    """Action to delete resource records from Route 53 hosted zones.
+
+    It is recommended to use a filter to avoid unwanted deletion
+    of R53 records from all hosted zones.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: route53-remove-filtered-records
+                resource: aws.rrset
+                filters:
+                  - type: value
+                    key: AliasTarget.DNSName
+                    value: "email.gc.example.com."
+                actions:
+                  - type: delete
+
+    """
+    schema = type_schema('delete',)
+    permissions = ('route53:ChangeResourceRecordSets',)
+    keys = (
+        'Name', 'Type', 'TTL', 'SetIdentifier', 'Region', 'AliasTarget', 'ResourceRecords')
+
+    def process(self, recordsets):
+        client = local_session(self.manager.session_factory).client('route53')
+        try:
+            for rrset in recordsets:
+
+                # Exempt the two zone associated mandatory records
+                if rrset['Type'] in ('NS', 'SOA'):
+                    continue
+
+                rrsetdata = self.generate_rrset(rrset)
+                self.manager.retry(
+                    client.change_resource_record_sets,
+                    HostedZoneId=rrset['c7n:parent-id'],
+                    ChangeBatch={
+                        'Changes': [
+                            {
+                                'Action': 'DELETE',
+                                'ResourceRecordSet': rrsetdata,
+                            }
+                        ]
+                    },
+                    ignore_err_codes=('InvalidChangeBatch'))
+        except Exception as e:
+                self.log.warning(
+                    "ResourceRecordSet delete error: %s", e)
+
+    def generate_rrset(self, recordset):
```

Comment:
ideally just make this a standalone function so the implementation isn't duplicated between the two classes.

Reply:
As asked I have made the **generate_rrset** method as standalone function